scripts/generate_gff3_db.py

```diff
@@ -0,0 +1,134 @@
+    "type TEXT, "
+    "start INT, "
+    "end INT, "
+    "score REAL, "
+    "strand TEXT, "
+    "name TEXT,"
+    "gene_name TEXT,"
+    "transcript_name TEXT,"
+    "attributes TEXT);")
+
+
+def _db_serialize(pyData):
+    return json.dumps(pyData, separators=(',', ':'))
+
+
+class Gff32Db(object):
+    """
+    Represents a unit of work for this script: Parse a GFF3 file
+    using an external GFF3 parser, and create a corresponding SQLite DB file
+    by iterating through the resulting parsed dictionary object
+    (gff3Data.byfeatureName).
+    """
+    def __init__(self, inputFile, outputFile):
+        """
+        :param inputFile: source GFF3 filename (can be a full path)
+        :param outputFile: destination sqlite filename (ditto)
+        """
+        self.gff3File = inputFile
+        self.dbFile = outputFile
+        if os.path.exists(outputFile):
+            print("DB output file already exists, please remove or rename.",
+                  file=sys.stderr)
+            exit()
+
+    def run(self):
+        print("Running GFF3 parser...", file=sys.stderr)
+        gff3Data = gff3.Gff3Parser(self.gff3File).parse()
+
+        print("Generating database...", file=sys.stderr)
+        dbconn = sqlite3.connect(self.dbFile)
+        dbcur = dbconn.cursor()
+        dbcur.execute(_dbTableSQL)  # create table
+        dbcur.execute("PRAGMA journal_mode=WAL")
+        dbconn.commit()
+
+        for i, featureName in enumerate(gff3Data.byFeatureName.keys()):
+            for feature in gff3Data.byFeatureName[featureName]:
+                # Ignores any parent IDs besides the first one.
+                parentIds = [parent.uniqueId for parent in feature.parents]
+                parentId = parentIds[0] if len(parentIds) > 0 else ''
+                # FIXME: No current code to ensure childIDs in correct order.
+                childIds = [child.uniqueId for child in feature.children]
+                values = (
+                    feature.uniqueId,
+                    parentId,
+                    _db_serialize(childIds),  # FIXME childIds
+                    feature.seqname,
+                    feature.source,
+                    feature.type,
+                    feature.start,
+                    feature.end,
+                    feature.score,
+                    feature.strand,
+                    feature.featureName,
+                    feature.attributes.get("gene_name", [None])[0],
+                    feature.attributes.get("transcript_name", [None])[0],
+                    _db_serialize(feature.attributes))
+                dbcur.execute((
+                    "INSERT INTO FEATURE ("
+                    "id, parent_id, child_ids, reference_name, "
+                    "source, type, start, end, score, strand, "
+                    "name, gene_name, transcript_name, attributes) "
+                    "VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?)"), values)
+                dbconn.commit()
+        dbcur.close()
```

Comment:
Make and verify indices in this script somewhere. Gencode.v.24 is about 2.8GB.

`create INDEX idx1 on feature(start, end, reference_name);`
`PRAGMA INDEX_LIST('feature');`